pal/src/tickcounter_sl.c

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include <stdlib.h>
+#include "azure_c_shared_utility/gballoc.h"
+
+#include <stdint.h>
+#include <time.h>
+#include "azure_c_shared_utility/tickcounter.h"
+#include "azure_c_shared_utility/optimize_size.h"
+#include "azure_c_shared_utility/xlogging.h"
+
+#define INVALID_TIME_VALUE      (time_t)(-1)
+typedef struct TICK_COUNTER_INSTANCE_TAG
+{
+    time_t init_time_value;
+    tickcounter_ms_t current_ms;
+} TICK_COUNTER_INSTANCE;
```

Comment:
@rmassand I don't understand why most ports use this `current_ms` value, since it's only ever written as a temporary value, and we apparently only expect time with a resolution of seconds, not milliseconds. But this isn't really a critique of TI's port, just a question about our API in general.

Reply:
ultimately all the time is boiled down to ms, but windows queries the system in nanoseconds and linux in seconds (which will eventually change to nanoseconds too). hence the name

---
In reply to: [227862179](https://github.com/massand/review_pal/pull/1#discussion_r227862179) [](ancestors = 227862179)